.pre-commit-config.yaml

```diff
@@ -0,0 +1,26 @@
+repos:
+  - repo: https://github.com/psf/black
+    rev: 22.6.0
+    hooks:
+      - id: black
+        files: \.py$|\.pyi$
+        language_version: python3.8
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v2.7.1
+    hooks:
+      - id: prettier
+        files: \.js$|\.jsx$|\.ts$|\.tsx$
+  - repo: local
+    hooks:
+      - id: run-typecheck
+        name: Run typecheck
+        entry: yarn --cwd frontend tsc --noEmit
+        files: \.js$|\.jsx$|\.ts$|\.tsx$
+        language: node
+        pass_filenames: false
+      - id: run-eslint
+        name: Run run eslint
+        entry: cd frontend && ./frontend/node_modules/.bin/eslint --ext .js --ext .jsx --ext .ts --ext .tsx --ignore-pattern './frontend/src/autogen/*' --max-warnings 0 ./frontend/src
```

Comment:
Would it be possible here if we just call `make jslint`, otherwise we are kind of duplicating this logic?

Reply:
pre-commit only runs on files that are changed i.e. when entry equals `black` and the files changed are `a.py` and `b.py`, the pre-commit executes the command: `black a.py b.py`.

Reply:
ahh I see 👍  So the difference is that the `make jslint` runs on all files and the eslint command with pre-commit only runs on files that have been changed in the commit?

Reply:
Yes. Pre-commit is a little smareter and quicker.